src/gill/TrouserPattern.java

```diff
@@ -154,470 +154,146 @@ protected void createBlocks()
-        else if (r_CrotchZ > 0)
-        {
-            stepFourteenY = ((v_SeatBkZ + r_CrotchZ) - (2.5 * Arb_CrotchReduction));
-        }
-        else if (r_CrotchZ == 0)
-        {
-            stepFourteenY = ((v_SeatBkZ) - (2.5 * Arb_CrotchReduction));
-        }
+        double centreXPosition = 0;
+        double centreYPosition = 0;
 
-        fullBlock.addKeypoint(new Vector2D((-(k_BodyRise - j_WaistToSeat)), stepFourteenY));
+        double crotchXPosition = centreXPosition;
+        double kneeXPosition = centreXPosition + n_CrotchHeight-m_KneeCRHeight;
+        double ankleXPosition = n_CrotchHeight- l_AnkleCRHeight;
+        double hipXPosition = -(o_HipCHeight-n_CrotchHeight);
+        double seatXPosition = -(p_SeatCHeight-n_CrotchHeight);
+        double waistXPosition = -k_BodyRise;
 
-        // Adding Step 15
-        // Has an if statement depending on the sign of measurement [047]
-        double stepFifteenY = 0.0;
-        if (r_CrotchZ < 0)
-        {
-            stepFifteenY = ((t_HipBkZ - r_CrotchZ) - (2.5 * Arb_CrotchReduction));
-        }
-        else if (r_CrotchZ > 0)
-        {
-            stepFifteenY = ((t_HipBkZ + r_CrotchZ) - (2.5 * Arb_CrotchReduction));
-        }
-        else if (r_CrotchZ == 0)
-        {
-            stepFifteenY = ((t_HipBkZ) - (2.5 * Arb_CrotchReduction));
-        }
+        double halfWaistSuppression = ((h_FrHipArc+i_BkHipArc)/2)-((e_FrWaistArc+f_BkWaistArc)/2);
 
-        fullBlock.addKeypoint(new Vector2D((-(k_BodyRise - a_WaistToHip)), stepFifteenY));
+        ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        /* Back half points, curves and dart */
+        ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-        // Adding Step 16
-        fullBlock.addKeypoint(new Vector2D(Arb_CrotchPointFive, 0.0));
+        //pre-calculation of point 7 y coordinate - used in a lot of calculations
+        double y7 = (i_BkHipArc/2)-t_HipBkZ-2.5+2;
+
+        Vector2D point1 = new Vector2D(crotchXPosition+0.5, centreYPosition);
+        Vector2D point2 = new Vector2D(kneeXPosition, y7/2-(((c_KneeCircR/2)+7+2)/2));
+        Vector2D point3 = new Vector2D(ankleXPosition,y7/2-(((d_AnkleCircleR/2)+10+2)/2));
+        Vector2D point5 = new Vector2D(ankleXPosition,y7/2+(((d_AnkleCircleR/2)+10+2)/2));
+        Vector2D point6 = new Vector2D(kneeXPosition, y7/2+(((c_KneeCircR/2)+7+2)/2));
+        Vector2D point7 = new Vector2D(crotchXPosition,y7);
+        Vector2D point8 = new Vector2D(hipXPosition,(i_BkHipArc/2)-t_HipBkZ-2.5+2);
+        Vector2D point9 = new Vector2D(seatXPosition,((-v_SeatBkZ)-2.5)+(g_BkSeatArc/2)+2);
+        Vector2D point10 = new Vector2D(waistXPosition,((-v_SeatBkZ)-2.5+(-x_WaistBkZ)-2.5)+(0.32*halfWaistSuppression)+(f_BkWaistArc/2)+0.35);
+        //todo: need to move point 11 in -x direction based on back crotch length
```

Comment:
Todo is usually written in capitals and leave a blank line before and after so it is clear. Not all IDEs highlight TODO lines.